addon/src/modifiers/on-key.js

```diff
@@ -1,111 +1,225 @@
 import Modifier from 'ember-modifier';
 import { inject as service } from '@ember/service';
 import { action } from '@ember/object';
+import { registerDestructor } from '@ember/destroyable';
+import { macroCondition, dependencySatisfies } from '@embroider/macros';
 import listenerName from '../utils/listener-name';
 import isKey from '../utils/is-key';
 
 const ONLY_WHEN_FOCUSED_TAG_NAMES = ['input', 'select', 'textarea'];
 
-/* This is an element modifier to trigger some behavior when
- * specified key combo is pressed. When used with a form element
- * (input, textarea, or select), the action fires only when element
- * has focus. When used with another element type, it will trigger the
- * passed action, OR if no action is passed, it will trigger a `click`
- * on the element. This allows for easy declaration of keyboard shortcuts
- * for anything clickable: In the following example, we trigger a
- * click on the button when the B key is pressed:
- *
- * <button
- *    type="button"
- *    {{on-key 'b'}}>
- *   Click me, or press "B"
- * </button>
- */
-export default class OnKeyModifier extends Modifier {
-  @service keyboard;
-
-  keyboardPriority = 0;
-  activatedParamValue = true;
-  eventName = 'keydown';
-  onlyWhenFocused = true;
-  listenerName;
-
-  didReceiveArguments() {
-    let [keyCombo, callback] = this.args.positional;
-    let { activated, event, priority } = this.args.named;
-    this.keyCombo = keyCombo;
-    this.callback = callback;
-    this.eventName = event || 'keydown';
-    this.activatedParamValue = Object.keys(this.args.named).includes(
-      'activated'
-    )
-      ? !!activated
-      : undefined;
-    this.keyboardPriority = priority ? parseInt(priority, 10) : 0;
-    this.listenerName = listenerName(this.eventName, this.keyCombo);
-    if (this.args.named.onlyWhenFocused !== undefined) {
-      this.onlyWhenFocused = this.args.named.onlyWhenFocused;
-    } else {
-      this.onlyWhenFocused = ONLY_WHEN_FOCUSED_TAG_NAMES.includes(
-        this.element.tagName.toLowerCase()
-      );
-    }
-  }
-
-  didInstall() {
-    this.keyboard.register(this);
-    if (this.onlyWhenFocused) {
-      this.element.addEventListener('click', this.onFocus, true);
-      this.element.addEventListener('focus', this.onFocus, true);
-      this.element.addEventListener('focusout', this.onFocusOut, true);
+let modifier;
+
+if (macroCondition(dependencySatisfies('ember-modifier', '>=3.2.0 || 4.x'))) {
```

Comment:
This increases code complexity.

However, my thought process is following:
* with such condition, only one version of the modifier class gets into bundle.
* we'll drop older version of ember-modifier support at some point, so one branch will just be deleted
* and most important IMO, it would be messy to use either `modify` *or* `didReceiveArguments`/`didInstall`/`willRemove` in the same class.
Hence having two separate classes is simpler to handle.